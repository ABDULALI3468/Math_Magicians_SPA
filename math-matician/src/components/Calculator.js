// eslint-disable-next-line
import React from "react";
import "../App.css";
class Calculator extends React.Component {
  render() {
    return (
      <div className="container">
        <div className="display">
          <input id="screen" type="text" placeholder="0" />
        </div>

        <div className="btns">
          <div className="row">
            <button id="ce">CE</button>
            <button>x!</button>
            <button className="btn">(</button>
            <button className="btn">)</button>
            <button className="btn">%</button>
            <button id="ac">AC</button>
          </div>

          <div className="row">
            <button>sin</button>
            <button>π</button>
            <button className="btn">7</button>
            <button className="btn">8</button>
            <button className="btn">9</button>
            <button className="btn">÷</button>
          </div>

          <div className="row">
            <button>cos</button>
            <button>log</button>
            <button className="btn">4</button>
            <button className="btn">5</button>
            <button className="btn">6</button>
            <button className="btn">×</button>
          </div>

          <div className="row">
            <button>tan</button>
            <button>√</button>
            <button className="btn">1</button>
            <button className="btn">2</button>
            <button className="btn">3</button>
            <button className="btn">-</button>
          </div>

          <div className="row">
            <button>e</button>
            <button>x 
              <span
               style={{position: "relative", bottom: ".4em", right: ".1em"}}>y</span>
            </button>
            <button className="btn">0</button>
            <button className="btn">.</button>
            <button id="eval">=</button>
            <button className="btn">+</button>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
