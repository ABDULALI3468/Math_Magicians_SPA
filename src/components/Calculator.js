import React from 'react';
import '../App.css';
import Button from './button';
import InputResult from './inputDisplay';
import calculate from '../logic/calculate';

class Calculator extends React.PureComponent {
  constructor(props) {
    super(props);
    this.onButtonClick = this.onButtonClick.bind(this);
    this.state = {
      obj: {
        total: null,
        next: null,
        operation: null,
      },
    };
  }

  onButtonClick = (e) => {
    const buttonName = e;
    const { obj } = this.state;
    this.setState({
      obj: calculate(obj, buttonName),
    });
  };

  render() {
    const { obj } = this.state;
    return (
      <div className="display">
        <InputResult displayState={obj} id="screen" />
        <Button buttonClick={this.onButtonClick} />
      </div>
    );
  }
}

export default Calculator;
