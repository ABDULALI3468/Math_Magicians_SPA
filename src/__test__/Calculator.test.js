import React from "react";
import renderer from "react-test-renderer";
import Calculator from "../components/Calculator";
import "@testing-library/jest-dom";

describe("Test of Calculator page", () => {
  it("test if Calculator page is rendered", () => {
    const calculator = renderer.create(<Calculator />).toJSON();
    expect(calculator).toMatchSnapshot();
  });
});
