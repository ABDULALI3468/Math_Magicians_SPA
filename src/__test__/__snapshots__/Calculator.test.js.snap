// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Calculator page test if Calculator page is rendered 1`] = `
<div
  className="calc_container"
>
  <div
    className="heading"
  >
    <p
      className="p1"
    >
      Lets Do
    </p>
    <p
      className="p2"
    >
      Some Maths
    </p>
  </div>
  <div
    className="container"
  >
    <div
      id="screen"
    >
      0
    </div>
    <div
      className="container"
    >
      <div
        className="btns"
      >
        <div
          className="row"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            (
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            )
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            id="ac"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            ร
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            id="eval"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
